neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,422 @@
+        {
+            UInt160 hash_account = new UInt160(account);
+            if (!engine.Service.CheckWitness(engine, hash_account)) return false;
+            StorageKey key_account = CreateStorageKey(ScriptHash, Prefix_Account, account);
+            if (engine.Service.Snapshot.Storages.TryGet(key_account) is null) return false;
+            StorageItem storage_account = engine.Service.Snapshot.Storages.GetAndChange(key_account);
+            AccountState state_account = AccountState.FromByteArray(storage_account.Value);
+            foreach (ECPoint pubkey in state_account.Votes)
+            {
+                StorageItem storage_validator = engine.Service.Snapshot.Storages.GetAndChange(CreateStorageKey(ScriptHash, Prefix_Validator, pubkey.ToArray()));
+                ValidatorState state_validator = ValidatorState.FromByteArray(storage_validator.Value);
+                state_validator.Votes -= state_account.Balance;
+                storage_validator.Value = state_validator.ToByteArray();
+            }
+            pubkeys = pubkeys.Distinct().Where(p => engine.Service.Snapshot.Storages.TryGet(CreateStorageKey(ScriptHash, Prefix_Validator, p.ToArray())) != null).ToArray();
+            if (pubkeys.Length != state_account.Votes.Length)
+            {
+                StorageItem storage_count = engine.Service.Snapshot.Storages.GetAndChange(CreateStorageKey(ScriptHash, Prefix_ValidatorsCount), () => new StorageItem
+                {
+                    Value = new ValidatorsCountState().ToByteArray()
+                });
+                ValidatorsCountState state_count = ValidatorsCountState.FromByteArray(storage_count.Value);
+                if (state_account.Votes.Length > 0)
+                    state_count.Votes[state_account.Votes.Length - 1] -= state_account.Balance;
+                if (pubkeys.Length > 0)
+                    state_count.Votes[pubkeys.Length - 1] += state_account.Balance;
+                storage_count.Value = state_count.ToByteArray();
+            }
+            state_account.Votes = pubkeys;
+            storage_account.Value = state_account.ToByteArray();
+            foreach (ECPoint pubkey in state_account.Votes)
+            {
+                StorageItem storage_validator = engine.Service.Snapshot.Storages.GetAndChange(CreateStorageKey(ScriptHash, Prefix_Validator, pubkey.ToArray()));
+                ValidatorState state_validator = ValidatorState.FromByteArray(storage_validator.Value);
+                state_validator.Votes += state_account.Balance;
+                storage_validator.Value = state_validator.ToByteArray();
+            }
+            return true;
+        }
+
+        private static ECPoint[] GetValidators(ApplicationEngine engine)
+        {
+            StorageItem storage_count = engine.Service.Snapshot.Storages.TryGet(CreateStorageKey(ScriptHash, Prefix_ValidatorsCount));
+            if (storage_count is null) return Blockchain.StandbyValidators;
```

Comment:
After call `initialize` we have the default validators stored (https://github.com/neo-project/neo/pull/717/files#diff-5b751f4efcbbbe0f62a66577c4f3ea95R252), but we don't store this key (`Prefix_ValidatorsCount`), so if we call `GetValidators` we don't read this validators

Reply:
I think your tests have answered your question. 😆

`Prefix_ValidatorsCount` is the vote statistics for validators count.

Reply:
But if i register a new validator, i won't be able to see this new validator if i don't vote, this is the expected behaviour?

Reply:
What do you mean "see this new validator"?

Reply:
@shargon, for the validator to enter someone need to vote in it before, right?
Thus, votes should be already registered for that holder.
I think that you will need to vote again for including this new one in your list.

I think that after this native NEO and GAS proceed with the native voting will be a great gain.